specs/management.yml

```diff
@@ -4479,6 +4479,38 @@ paths:
       name: id
       in: path
       required: true
+  /health-check:
+    get:
+      security: []
+      description: Returns OK or error depending on controller health status
+      tags:
+      - Informational
+      summary: Returns OK or error depending on controller health status
+      operationId: healthCheck
+      responses:
+        "200":
+          description: Base empty response
+          schema:
+            $ref: '#/definitions/empty'
```

Comment:
I doubt this returns an empty response. Needs to point to the schema of whatever is returned. Useful for generated clients and testing.

Reply:
Also the resulting `rest_model.*` structs should be used in `controller/internal/routes/health_check_router.go` to format the output.

Reply:
It's returning as empty of a body as possible (using responder.RespondWithEmptyOk()). If everything is ok, it returns status code 200. If there's a problem it returns 503 with the error output from the health checker. I had the error bits encoded incorrectly, that's fixed now.

Reply:
Are you basing this pattern off something, it seems weird. I would expect a 200 ok to return a payload of 0 or more issues or the actual stats themselves.

Some frameworks that are generated from specs treat 4xx and 5xx responses as exceptions that must be caught. Seems weird to get an exception to catch it then look at what the payload is.

Reply:
The health check library comes with an http serving library, which is used in the router. I didn't want to introduce a third potential endpoint for the controller so I added it to the management API. The built in API return 200 on healthy and 503 on unhealthy. It does output full check data on both success and fail. We only have one check right now so I didn't want to spend the time. I figured we can add more detailed output if add additional checks.

Reply:
4xx and 5xx errors are meant to convey that the request could not be completed for some reason. 4xx for client correctable issues, 5xx for server-side issues.

Using 503 is not correct and in my opinion, would be unexpected.

If we only have 1 check  and the only viable output response is "good" or "bad", I suggest this response payload:

```
{
   "healthy": true/false
}
```

Reply:
I pulled the branch to see what the health check is doing. Seems pretty trivial to model their results generically.


```
responses:
  listHealthChecks:
    description: A list of controler health checks
    schema:
      $ref: '#/definitions/listHealthCheckEnvelope'

definitions:
   listHealthCheckEnvelope:
    type: object
    required:
      - meta
      - data
    properties:
      meta:
        $ref: '../shared/standard-responses.yml#/definitions/meta'
      data:
        $ref: '#/definitions/healthCheckResultList'
   healthCheckResultList:
      type: array
      items:
         $ref: '#/definitions/healCheckResultDetail'
   healthCheckResultDetail:
     type: object
     properties:
       message:
         type: string
       details:
          type: object
          additionalProperties: {}
        error:
          type: string
        timestamp:
          type: string
          format: date-time
       duration:
          type: string
          format: date-time
       contigousFailures:
          type: integer
       timeOfFirstFailure:
          type: string
          format: date-time
```